solidity/src/base/Transcript.sol

```diff
@@ -0,0 +1,202 @@
+// SPDX-License-Identifier: UNLICENSED
+// This is licensed under the Cryptographic Open Software License 1.0
+pragma solidity ^0.8.28;
+
+import {MODULUS, WORD_SIZE} from "./Constants.sol";
+// assembly only constants
+// solhint-disable-next-line no-unused-import
+import {MODULUS_MASK, FREE_PTR} from "./Constants.sol";
+
+/// A library for managing a simple public coin transcript.
+/// The transcript is a sequence of messages and challenges, where each challenge is a scalar (less than MODULUS).
+library Transcript {
+    /// Draw a challenge from the transcript, and update the state of the transcript.
+    /// The result should be a scalar value (less than MODULUS).
+    /// The challenges is a masked version of the current state of the transcript.
+    /// The new state of the transcript is the hash of the current state.
+    ///
+    /// In particular, the critical property is that the challenges are deterministic and uniformly distributed,
+    /// but with no collisions.
+    function drawChallenge(uint256[1] memory transcript_) private pure returns (uint256 result_) {
+        assembly {
+            function draw_challenge(transcript_ptr) -> result {
+                result := and(mload(transcript_ptr), MODULUS_MASK)
```

Comment:
since that MODULUS_MASK is always ones, do we still need to apply AND operation? 

result := mload(transcript_ptr)